projects/kit/components/input-number/input-number.component.ts

```diff
@@ -0,0 +1,255 @@
+import {
+    ChangeDetectionStrategy,
+    Component,
+    computed,
+    inject,
+    Input,
+    signal,
+} from '@angular/core';
+import {toSignal} from '@angular/core/rxjs-interop';
+import {MaskitoDirective} from '@maskito/angular';
+import type {MaskitoOptions} from '@maskito/core';
+import {maskitoInitialCalibrationPlugin} from '@maskito/core';
+import {
+    maskitoCaretGuard,
+    maskitoNumberOptionsGenerator,
+    maskitoParseNumber,
+} from '@maskito/kit';
+import {TuiControl} from '@taiga-ui/cdk/classes';
+import {CHAR_HYPHEN, CHAR_MINUS} from '@taiga-ui/cdk/constants';
+import {TUI_IS_IOS, tuiFallbackValueProvider} from '@taiga-ui/cdk/tokens';
+import {tuiInjectElement} from '@taiga-ui/cdk/utils/dom';
+import {tuiIsSafeToRound} from '@taiga-ui/cdk/utils/math';
+import {TuiTextfieldDirective} from '@taiga-ui/core/components/textfield';
+import {TUI_DEFAULT_NUMBER_FORMAT, TUI_NUMBER_FORMAT} from '@taiga-ui/core/tokens';
+import {tuiFormatNumber} from '@taiga-ui/core/utils/format';
+import {tuiMaskito} from '@taiga-ui/kit/utils';
+
+import {TUI_INPUT_NUMBER_OPTIONS} from './input-number.options';
+
+const DEFAULT_MAX_LENGTH = 18;
+
+@Component({
+    standalone: true,
+    selector: 'input[tuiInputNumber]',
+    template: '',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [tuiFallbackValueProvider(null)],
+    hostDirectives: [
+        // TODO: replace with `TuiWithTextfield` after merging of https://github.com/taiga-family/taiga-ui/pull/9976
```

Comment:
Merged

Reply:
Fixed in https://github.com/taiga-family/taiga-ui/pull/10031/commits/21703a58cae39021aef32a9ecf4cb297188a55ec